### ISSUE TITLE
hiredis: External LLVMFunction redisConnectNonBlock cannot be found. (LLVMLinkerException)

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1974]
```
truffleruby 20.1.0-dev-74a40b57, like ruby 2.6.5, GraalVM CE Native [x86_64-darwin]
```

To reproduce:

```
gem install hiredis
ruby -rhiredis -e 'Hiredis::Connection.new.connect("127.0.0.1", 6379)'
```

Note: redis needs to be installed and be run

Error:

```
~/.gem/truffleruby/2.6.5/gems/hiredis-0.6.3/ext/hiredis_ext/connection.c:261:in `connection_connect': External LLVMFunction redisConnectNonBlock cannot be found. (LLVMLinkerException) (RuntimeError)
Translated to internal error
	from ~/.rubies/truffleruby-dev/lib/truffle/truffle/cext_ruby.rb:39:in `connect'
	from -e:1:in `<main>'
```

**Comment by eregon** [comment]
This happens most likely because hiredis is built as a static library and doesn't use `RbConfig::CONFIG['CC']` to build hiredis but the system compiler:
https://github.com/redis/hiredis-rb/blob/master/ext/hiredis_ext/extconf.rb

I'm thinking to add a TruffleRuby option to add the toolchain `bin` on PATH and set env vars CC/CXX on `require 'mkmf'` or at least when running the `extconf.rb`.

**Comment by deepj** [comment]
Closed due to I'm not welcome anymore in Ruby community

**Comment by eregon** [comment]
The idea above could potentially let us use the vendored libxml2/libxslt for nokogiri. cc @chrisseaton

**Comment by ivoanjo** [comment]
Hey @eregon we're bitten by this as well in the [`ddtrace`](https://github.com/DataDog/dd-trace-rb) integration tests.

Is there any working workaround, beyond just skipping any hiredis stuff when on TruffleRuby?

**Comment by eregon** [comment]
Using https://github.com/redis/redis-rb is the current workaround.
I should be able to work on this soon, I think it might be relatively easy to get hiredis working.

**Comment by eregon** [comment]
Prepending the toolchain to PATH seems the right thing to do for `hiredis`.
Doing so, the gem just works and the tests pass for me locally (except one failure which also happens on CRuby).
I'll make a PR to do that by default, and have an option to disable in case it's problematic for some gem.

Setting CC/CXX doesn't seem enough, notably because of https://github.com/redis/hiredis/blob/685030652cd98c5414ce554ff5b356dfe8437870/Makefile#L54
It does seem to work with the system `ar` rather than the toolchain `ar` but that just seems lucky and not particularly reliable.

**Comment by eregon** [comment]
Fixed in 5f3339321c6bf3f30e7993101b2f2abce7f1b0f4, now `hiredis` should work just fine on TruffleRuby!
